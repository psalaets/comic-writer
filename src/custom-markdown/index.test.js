import { parse } from './index';

describe('custom markdown', () => {
  describe('parse()', () => {
    it('single page', () => {
      const result = parse(`
## Page 1

`);

      expect(result).toMatchSnapshot();
    });

    it('two pages', () => {
      const result = parse(`
## Page 1

## Page 2

`);

      expect(result).toMatchSnapshot();
    });

    it('pages with panels', () => {
      const result = parse(`
## Page 1

## Page 2

### Panel 1

### Panel 2
`);

      expect(result).toMatchSnapshot();
    });

    it('pages with panels and lettering', () => {
      const result = parse(`
## Page 1

## Page 2

### Panel 1

> bob: ok

> bob: try it *again*

### Panel 2

This is panel description.

> sfx: boom

> caption: yep
`);

      expect(result).toMatchSnapshot();
    });
  });
});