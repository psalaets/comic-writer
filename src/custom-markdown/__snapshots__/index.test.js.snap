// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom markdown parse() pages with panels 1`] = `
Array [
  Object {
    "content": Array [],
    "type": "paragraph",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "1",
    "number": 1,
    "panelCount": 0,
    "sfxCount": 0,
    "type": "page",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [
      Object {
        "captionCount": 0,
        "captionWords": 0,
        "content": Array [],
        "dialogueCount": 0,
        "dialogueWords": 0,
        "id": "2.1",
        "number": 1,
        "sfxCount": 0,
        "type": "panel",
      },
      Object {
        "captionCount": 0,
        "captionWords": 0,
        "content": Array [],
        "dialogueCount": 0,
        "dialogueWords": 0,
        "id": "2.2",
        "number": 2,
        "sfxCount": 0,
        "type": "panel",
      },
    ],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "2",
    "number": 2,
    "panelCount": 2,
    "sfxCount": 0,
    "type": "page",
  },
]
`;

exports[`custom markdown parse() pages with panels and lettering 1`] = `
Array [
  Object {
    "content": Array [],
    "type": "paragraph",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "1",
    "number": 1,
    "panelCount": 0,
    "sfxCount": 0,
    "type": "page",
  },
  Object {
    "captionCount": 1,
    "captionWords": 1,
    "content": Array [
      Object {
        "captionCount": 0,
        "captionWords": 0,
        "content": Array [
          Object {
            "content": Array [
              Object {
                "content": "ok",
                "type": "text",
              },
            ],
            "id": "2.1.1",
            "modifier": undefined,
            "number": 1,
            "speaker": "bob",
            "type": "dialogue",
            "wordCount": 1,
          },
          Object {
            "content": Array [
              Object {
                "content": "try it ",
                "type": "text",
              },
              Object {
                "content": Array [
                  Object {
                    "content": "again",
                    "type": "text",
                  },
                ],
                "type": "em",
              },
            ],
            "id": "2.1.2",
            "modifier": undefined,
            "number": 2,
            "speaker": "bob",
            "type": "dialogue",
            "wordCount": 3,
          },
        ],
        "dialogueCount": 2,
        "dialogueWords": 4,
        "id": "2.1",
        "number": 1,
        "sfxCount": 0,
        "type": "panel",
      },
      Object {
        "captionCount": 1,
        "captionWords": 1,
        "content": Array [
          Object {
            "content": Array [
              Object {
                "content": "This is panel description",
                "type": "text",
              },
              Object {
                "content": ".",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
          Object {
            "content": Array [],
            "type": "paragraph",
          },
          Object {
            "content": Array [
              Object {
                "content": "boom",
                "type": "text",
              },
            ],
            "id": "2.2.3",
            "modifier": undefined,
            "number": 3,
            "type": "sfx",
          },
          Object {
            "content": Array [
              Object {
                "content": "yep",
                "type": "text",
              },
            ],
            "id": "2.2.4",
            "modifier": undefined,
            "number": 4,
            "type": "caption",
            "wordCount": 1,
          },
        ],
        "dialogueCount": 0,
        "dialogueWords": 0,
        "id": "2.2",
        "number": 2,
        "sfxCount": 1,
        "type": "panel",
      },
    ],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "2",
    "number": 2,
    "panelCount": 2,
    "sfxCount": 1,
    "type": "page",
  },
]
`;

exports[`custom markdown parse() single page 1`] = `
Array [
  Object {
    "content": Array [],
    "type": "paragraph",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "1",
    "number": 1,
    "panelCount": 0,
    "sfxCount": 0,
    "type": "page",
  },
]
`;

exports[`custom markdown parse() two pages 1`] = `
Array [
  Object {
    "content": Array [],
    "type": "paragraph",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "1",
    "number": 1,
    "panelCount": 0,
    "sfxCount": 0,
    "type": "page",
  },
  Object {
    "captionCount": 0,
    "captionWords": 0,
    "content": Array [],
    "dialogueCount": 0,
    "dialogueWords": 0,
    "id": "2",
    "number": 2,
    "panelCount": 0,
    "sfxCount": 0,
    "type": "page",
  },
]
`;
